src/screens/MarketDetailScreen/MarketDetailPage.tsx

```diff
@@ -0,0 +1,206 @@
+      if (scrollViewRef.current && sectionOffsets[tag] !== undefined) {
+        scrollViewRef.current.scrollTo({
+          x: 0,
+          y: sectionOffsets[tag],
+          animated: true,
+        });
+      }
+    },
+    [sectionOffsets],
+  );
+
+  const scrollToSidebarTag = useCallback(
+    (tag: string) => {
+      if (tagScrollViewRef.current) {
+        tagScrollViewRef.current.scrollTo({
+          x: 100 * Object.keys(productsByTags).indexOf(tag),
+          animated: true,
+        });
+      }
+    },
+    [productsByTags],
+  );
+
+  const handleScroll = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+    const contentOffsetY = event.nativeEvent.contentOffset.y;
+    Object.keys(sectionOffsets).forEach(tag => {
+      if (
+        contentOffsetY >= sectionOffsets[tag] &&
+        contentOffsetY < sectionOffsets[tag] + 100
+      ) {
+        setSelectedTag(tag);
+      }
+    });
+  };
+  const updateSectionOffsets = () => {
+    let offsetY = 0;
+    const newOffsets: {[key: string]: number} = {};
+    Object.keys(productsByTags).forEach(tag => {
+      newOffsets[tag] = offsetY;
+      offsetY += 200;
+    });
+    setSectionOffsets(newOffsets);
+  };
+
+  const handleLayout = (tag: string) => (event: LayoutChangeEvent) => {
+    const {y} = event.nativeEvent.layout;
+    setSectionOffsets(prevOffsets => ({
+      ...prevOffsets,
+      [tag]: y,
+    }));
+  };
+
+  useEffect(() => {
+    if (selectedTag) {
+      scrollToSidebarTag(selectedTag);
+    }
+  }, [selectedTag, scrollToSidebarTag]);
+
+  return (
+    <S.MarketDetailInfoView>
+      {/* <MarketImageSlider /> */}
+      <S.MarketMainInfoWrapper>
+        <S.MarKetName>{name} </S.MarKetName>
+        <S.MarketDescription>
+          내 자식에게 준다는 마음으로 음식을 만들고 있습니다^^
+        </S.MarketDescription>
+      </S.MarketMainInfoWrapper>
+      <S.MarketSideInfoWrapper>
+        <S.MarketSideInfo>{`픽업: ${date.format(
+          pickupStartAt,
+          'HH시 mm분',
+        )} ~ ${date.format(pickupEndAt, 'HH시 mm분')}`}</S.MarketSideInfo>
+        <S.MarketSideInfo>{address}</S.MarketSideInfo>
+      </S.MarketSideInfoWrapper>
+      <S.SideTagBarScrollView
+        horizontal
+        showsHorizontalScrollIndicator={false}
+        ref={tagScrollViewRef}>
+        {Object.keys(productsByTags).map(tag => (
+          <TouchableOpacity
+            key={tag}
+            onPress={() => {
+              setSelectedTag(tag);
+              scrollToSection(tag);
+            }}>
+            <S.SideBarView selected={selectedTag === tag}>
+              <S.SideBarText selected={selectedTag === tag}>
+                {tag}
+              </S.SideBarText>
+            </S.SideBarView>
+          </TouchableOpacity>
+        ))}
+      </S.SideTagBarScrollView>
+
+      <S.Divider />
+
+      <S.MenuScrollView
+        ref={scrollViewRef}
+        onScroll={handleScroll}
+        onLayout={() => updateSectionOffsets()}>
+        {Object.keys(productsByTags).map(tag => (
```

Comment:
```suggestion
        onScroll={handleScroll}
        onLayout={updateSectionOffsets}>
        {Object.keys(productsByTags).map(tag => (
```
익명 함수 없이 함수명을 바로 넘겨줘도 괜찮지 않나요?